### ISSUE TITLE
NAMESPACE error starting httpd related to /var/tmp being a symlink

### ISSUE THREAD

**Issue by chesty** [freeipa/freeipa-container#254]
Hi, love your work.

Yesterday I rebuilt a server with zfs and had trouble starting my freeipa container, it took a little digging around because I assumed it was zfs related, and there were some issues relating to posixacl and whatnot, but thanks to murphy I think there was a patch 3 days ago to make /var/tmp a symlink to /tmp/var/tmp?

If you duck or quant around for "Failed at step NAMESPACE spawning" you'll probably see some of the same posts I read pointing to systemd and /var/tmp being a symlnk. I don't have any idea why it matters, but for now I patch /etc/volume-tmp-list to remove /var/tmp and I was able to start freeipa (and I love freeipa and docker too, thanks)

go any ideas? It's not a bug with freeipa or the freeipa containers as it obviously works for you, it's probably something related to my ubuntu host, although many of the posts I read with the NAMESPACE error were from fedora.

**Comment by adelton** [comment]
I've seen services that have `PrivateTmp=yes` failing with

```
      [error] CalledProcessError: CalledProcessError(Command ['/bin/systemctl', 'restart', 'ipa-custodia.service'] returned non-zero exit status 1: 'Job for ipa
-custodia.service failed because the control process exited with error code.\nSee "systemctl status ipa-custodia.service" and "journalctl -xe" for details.\n')
    with
    systemd[524]: ipa-custodia.service: Failed to set up mount namespacing: No such file or directory
    systemd[524]: ipa-custodia.service: Failed at step NAMESPACE spawning /usr/libexec/ipa/ipa-custodia: No such file or directory
```

when `/var/tmp` is a symlink, like it is in the latest code.

However, in 73c55af44951619cebcd7a6a959ce0846bff0ad5 I have removed any needs to run FreeIPA container as privileged and in fact, we now run the Travis checks on all OSes (in container) and on the Ubuntu that is in Travis without `--privileged`.

So, what is the exact `docker run` that you use to start the container? Does it have `--privileged` and will things improve when you remove it?

**Comment by chesty** [comment]
I don't believe I am running it as a privileged container. I do have SYS_ADMIN capability but I didn't make a note as to why I added it. OK, I need it so the ipa boot scripts can set the hostname amongst other things.

this is without SYS_ADMIN
```
ipa02_1_5af7622815bb | Failed to set hostname to <ipa02>: Operation not permitted
ipa02_1_5af7622815bb | Initializing machine ID from random generator.
ipa02_1_5af7622815bb | Couldn't move remaining userspace processes, ignoring: Input/output error
ipa02_1_5af7622815bb | system.slice: Failed to set invocation ID on control group /system.slice/containerd.service/system.slice, ignoring: Operation not permitted
ipa02_1_5af7622815bb | fedora-domainname.service: Failed to set invocation ID on control group /system.slice/containerd.service/system.slice/fedora-domainname.service, ignoring: Operation not permitted
ipa02_1_5af7622815bb | systemd-journald.service: Failed to set invocation ID on control group /system.slice/containerd.service/system.slice/systemd-journald.service, ignoring: Operation not permitted
```

Here's most of my docker-compose

```yaml
version: '3.4'
services:
  ipa02:
    image: freeipa/freeipa-server:fedora-27
    restart: always
    hostname: ipa02
    domainname: blah.com.au
    stop_signal: RTMIN+3
    tty: true
    stdin_open: true
    cap_add:
      - SYS_ADMIN
    sysctls:
      - net.ipv6.conf.all.disable_ipv6=0
    environment:
      - IPA_SERVER_HOSTNAME=ipa02.blah.com.au
      - IPA_SERVER_IP=192.168.1.51
      - DEBUG_TRACE=1
      - DEBUG_NO_EXIT=1
    security_opt:
      - apparmor=unconfined
      - seccomp=unconfined
    volumes:
      - type: bind
        source: /sys/fs/cgroup
        target: /sys/fs/cgroup
        read_only: true
      - type: volume
        source: ipa-data
        target: /data
    ports:
...
```

**Comment by adelton** [comment]
That `SYS_ADMIN` should not be necessary. The container does not need to change the hostname, `IPA_SERVER_HOSTNAME` should be enough. Can you try without `SYS_ADMIN`?

**Comment by adelton** [comment]
Also, why do you use the `seccomp=unconfined`? In our testing, that has always caused more problems than it solved (and we had to take special steps to workaround setups where people do this). Overall, `SYS_ADMIN` and `seccomp=unconfined` gets you very close to privileged.

**Comment by chesty** [comment]
I had a lot of problems trying to get with working when I first started using it, seccomp=unconfined was probably in some post I read and added it trying to get it working. I'll remove them but I can't test right now, but I'll be back soon. thanks.

**Comment by chesty** [comment]
I had a few things I needed to change.
```hostname: ipa02
domainname: blah.com.au```

I deleted domainname and made hostname ipa02.blah.com.au, then it no longer tried to set a hostname, so SYS_ADMIN was no longer needed (I noticed in my above logs with was trying to set a hostname of ipa02). 

After I removed SYS_ADMIN it started OK. I believe SYS_ADMIN also causes a problem with a /var/tmp symlink, I don't know if seccomp=unconfined does, I removed it and didn't test with it enabled and SYS_ADMIN removed.

Thank you very much. Much appreciated.
